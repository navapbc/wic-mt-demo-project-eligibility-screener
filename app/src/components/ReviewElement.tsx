import { Trans } from 'next-i18next'
import { ReactElement } from 'react'

import { i18nKey } from '@src/types'

export type ReviewElementProps = {
  labelKey: i18nKey
  children: ReactElement | string | null
}

const ReviewElement = (props: ReviewElementProps): ReactElement => {
  const { labelKey, children } = props

  return (
    <div className="review-element margin-bottom-3">
      <dt>
        <strong>
          <Trans i18nKey={labelKey} />
        </strong>
      </dt>
      <dd className="margin-left-0">{children}</dd>
    </div>
  )
}

export default ReviewElement
