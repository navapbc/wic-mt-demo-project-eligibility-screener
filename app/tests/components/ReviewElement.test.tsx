import { render, screen } from '@testing-library/react'

import ReviewElement from '@components/ReviewElement'

import { testSnapshot } from '../helpers/sharedTests'

it('should match snapshot', () => {
  const element = (
    <ReviewElement labelKey="label">This is a child</ReviewElement>
  )
  testSnapshot(element)
})

it('should render children', () => {
  render(<ReviewElement labelKey="label">This is a child</ReviewElement>)
  const child = screen.getByText('This is a child')
  expect(child).toBeInTheDocument()
})
