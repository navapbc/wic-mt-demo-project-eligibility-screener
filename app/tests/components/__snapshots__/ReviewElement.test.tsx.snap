// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="review-element margin-bottom-3"
>
  <dt>
    <strong>
      label
    </strong>
  </dt>
  <dd
    className="margin-left-0"
  >
    This is a child
  </dd>
</div>
`;
